lib/assert.js

```diff
@@ -167,8 +167,11 @@ function isObjectOrArrayTag(tag) {
 // a) The same built-in type tags
 // b) The same prototypes.
 function strictDeepEqual(actual, expected) {
-  if (actual === null || expected === null ||
-    typeof actual !== 'object' || typeof expected !== 'object') {
+  if (typeof actual !== 'object') {
+    return typeof actual === 'number' && Number.isNaN(actual) &&
```

Comment:
I think `typeof actual === 'number'` can be removed, `Number.isNaN(actual)` is sufficient.

Reply:
That is correct but it is faster for all other types to check for number first and if it is a number it is not a real penalty because that check is super fast.

Reply:
Does it makes sense to do the same for `expected` then?

Reply:
We could but I feel like that is overdoing it as it would only apply for the case where `Number.isNaN(actual)`, especially as I would expect `expected` to also be NaN in that case more often then not being of type number.